MeshLib/MeshEditing/Mesh2MeshPropertyInterpolation.cpp

```diff
@@ -45,71 +47,86 @@ bool Mesh2MeshPropertyInterpolation::setPropertiesForMesh(Mesh *dest_mesh, std::
         return false;
     }
 
-    GeoLib::AABB src_aabb(_src_mesh->getNodes().begin(), _src_mesh->getNodes().end());
-    GeoLib::AABB dest_aabb(dest_mesh->getNodes().begin(), dest_mesh->getNodes().end());
-    if (!src_aabb.containsAABB(dest_aabb)) {
-        ERR("MeshLib::Mesh2MeshPropertyInterpolation::setPropertiesForMesh() source mesh to small.");
-        ERR("src_aabb: %f, %f, %f | %f, %f, %f", src_aabb.getMinPoint()[0], src_aabb.getMinPoint()[1], src_aabb.getMinPoint()[2], src_aabb.getMaxPoint()[0], src_aabb.getMaxPoint()[1], src_aabb.getMaxPoint()[2]);
-        ERR("dest_aabb: %f, %f, %f | %f, %f, %f", dest_aabb.getMinPoint()[0], dest_aabb.getMinPoint()[1], dest_aabb.getMinPoint()[2], dest_aabb.getMaxPoint()[0], dest_aabb.getMaxPoint()[1], dest_aabb.getMaxPoint()[2]);
-        return false;
-    }
-
-    interpolatePropertiesForMesh(dest_mesh, dest_properties);
+    interpolatePropertiesForMesh(dest_mesh);
 
     return true;
 }
 
-void Mesh2MeshPropertyInterpolation::interpolatePropertiesForMesh(Mesh *dest_mesh, std::vector<double>& dest_properties) const
+void Mesh2MeshPropertyInterpolation::interpolatePropertiesForMesh(
+    Mesh *dest_mesh) const
 {
+    // check the existence of PropertyVector in the source mesh
+    boost::optional<MeshLib::PropertyVector<double> const&> opt_src_pv(
+        _src_mesh->getProperties().getPropertyVector<double>(_property_name));
+    if (!opt_src_pv) {
+        WARN("Did not find PropertyVector<double> \"%s\" in source mesh.",
+            _property_name.c_str());
+        return;
+    }
+
     // carry over property information from source elements to source nodes
     std::vector<double> interpolated_src_node_properties(_src_mesh->getNumberOfNodes());
     interpolateElementPropertiesToNodeProperties(interpolated_src_node_properties);
 
+    boost::optional<MeshLib::PropertyVector<double> &> opt_pv(
+        dest_mesh->getProperties().getPropertyVector<double>(_property_name));
+    if (!opt_pv) {
+        INFO("Create new PropertyVector \"%s\" of type double.",
+             _property_name.c_str());
+        opt_pv = dest_mesh->getProperties().createNewPropertyVector<double>(
+            _property_name, MeshItemType::Cell, 1);
+        if (!opt_pv) {
+            WARN("Could not get or create a PropertyVector of type double"
+                " using the given name \"%s\".", _property_name.c_str());
+            return;
+        }
+    }
+    MeshLib::PropertyVector<double> & dest_properties(opt_pv.get());
+    if (dest_properties.size() != dest_mesh->getNumberOfElements())
```

Comment:
If that can only happen upon newly created properties, I'd rather move the resize call upwards to the creation code.